src/diffusers/loaders.py

```diff
@@ -464,6 +479,36 @@ def save_function(weights, filename):
         save_function(state_dict, os.path.join(save_directory, weight_name))
         logger.info(f"Model weights saved in {os.path.join(save_directory, weight_name)}")
 
+    def _load_lora_aux(self, state_dict, network_alpha=None):
+        lora_grouped_dict = defaultdict(dict)
+        for key, value in state_dict.items():
+            attn_processor_key, sub_key = ".".join(key.split(".")[:-3]), ".".join(key.split(".")[-3:])
+            lora_grouped_dict[attn_processor_key][sub_key] = value
+
+        for key, value_dict in lora_grouped_dict.items():
+            rank = value_dict["lora.down.weight"].shape[0]
+            hidden_size = value_dict["lora.up.weight"].shape[0]
+            target_modules = [module for name, module in self.named_modules() if name == key]
+            if len(target_modules) == 0:
+                logger.warning(f"Could not find module {key} in the model. Skipping.")
+                continue
+
+            target_module = target_modules[0]
+            value_dict = {k.replace("lora.", ""): v for k, v in value_dict.items()}
+
+            lora = None
+            if isinstance(target_module, LoRACompatibleConv):
+                lora = LoRAConv2dLayer(hidden_size, hidden_size, rank, network_alpha)
+            elif isinstance(target_module, LoRACompatibleLinear):
+                lora = LoRALinearLayer(target_module.in_features, target_module.out_features, rank, network_alpha)
+            else:
+                raise ValueError(f"Module {key} is not a LoRACompatibleConv or LoRACompatibleLinear module.")
+            lora.load_state_dict(value_dict)
+            lora.to(device=self.device, dtype=self.dtype)
+
+            # install lora
+            target_module.lora_layer = lora
```

Comment:
I consider this degree of state dict munging to be an anti-pattern. What can we do to make the code as close as possible to looking like just loading a normal state dict? If we need to do any change to the model or the state dict, ideally they are done very clearly ahead of time instead of interspersed with eachother.

I.e. if necessary, this is the ideal case

```
state_dict = ...
model = ...

state_dict = convert_state_dict(state_dict)
model = convert_model(model)

model.load(state_dict)
```

Reply:
The auxiliary modules are not always present actually. So, all things considered (flexibility, simplicity, and readability), this is something acceptable to me. 

Would appreciate some examples and more thoughts on how to approach what you're suggesting.